### PR TITLE
fix docs speed

### PR DESCRIPTION
**Pull request by MildTomato**
Make the docs load pages faster.

Used react.memo in places. 
Moved state changes to be lower down to reduce re-rendering

### CODE REVIEW

**Review by silentworks — APPROVED** [review]

File: apps/docs/components/DocSearch.tsx [hunk 1]
```diff
@@ -115,6 +115,10 @@ export function SearchProvider({ children }: any) {
   )
 }
 
+const SearchProvider = memo(SearchProviderContent)
```
Comment (alaister) [inline apps/docs/components/DocSearch.tsx:118]: This memo is probably adding more overhead than it's saving. The `children` prop will probably change on every render, so this memo isn't effective.

Adding a memo around whatever you passing in as `children` will probably yield better results. For example:

```typescript
<SearchProvider>
  <MyComp /> // MyComp should be memo'd
</SearchProvider>
```

This way, when the state changes in the provider, MyComp won't rerender as its props don't change.

**Review by alaister — APPROVED** [review]
Looks like a significant improvement!

Left a comment about a small optimisation you can make regarding memoing a context, but it's not a big deal perf-wise if you want to skip it here.

There are also a few things commented out that could be cleaned up if you like!

File: apps/docs/components/DocSearch.tsx (hunk 1 above)
Comment (MildTomato) [reply apps/docs/components/DocSearch.tsx:118]: oh yea, this was from back when i was memoing everything, as i was trying to find what was the bottleneck.
I'll remove.

turns out the biggest offender was next router 🙄

Comment (MildTomato) [reply apps/docs/components/DocSearch.tsx]: removed

**Comment by MildTomato** [comment]
left to do is refactoring the nav code so it isn't a mess.
a lot of duplicate code, especially with lists of "menu items"

but can probably merge in now

**Pull request merged**